be2-scala/src/test/scala/ch/epfl/pop/pubsub/graph/handlers/GetMessagesByIdResponseHandlerSuite.scala

```diff
@@ -1,99 +1,85 @@
-      case DbActor.WriteAndPropagate(channel, message) =>
-        testProbe ! (channel, message)
-        sender() ! DbActor.DbActorAck()
+  // Implicit for system actors
+  implicit val timeout: Timeout = Timeout(1.seconds)
+
+  val inMemoryStorage: InMemoryStorage = InMemoryStorage()
+  val messageRegistry: MessageRegistry = MessageRegistry()
+  val pubSubMediatorRef: ActorRef = system.actorOf(PubSubMediator.props, "PubSubMediator")
+  val dbActorRef: AskableActorRef = system.actorOf(Props(DbActor(pubSubMediatorRef, messageRegistry, inMemoryStorage)), "DbActor")
+
+  // Inject dbActor above
+  PublishSubscribe.buildGraph(pubSubMediatorRef, dbActorRef, messageRegistry, ActorRef.noSender, ActorRef.noSender, isServer = false)
+
+  // Helper
+  // Get rid of decoded data to compare against original message
+  private def getMessages(channel: Channel): Set[Message] = {
+    val ask = dbActorRef ? DbActor.Catchup(channel)
+    Await.result(ask, duration) match {
+      case DbActor.DbActorCatchupAck(list) => list
+          .map(msg => Message(msg.data, msg.sender, msg.signature, msg.message_id, msg.witness_signatures, None)).toSet
+      case _ => Matchers.fail(s"Couldn't catchup on channel: $channel")
     }
   }
-  class FailingTestDb(testProbe: ActorRef) extends Actor {
-    override def receive: Receive = {
-      case DbActor.WriteAndPropagate(channel, message) =>
-        testProbe ! (channel, message)
-        sender() ! Status.Failure(DbActorNAckException(ErrorCodes.INVALID_ACTION.id, s"channel '$channel' does not exist in db"))
-    }
+
+  override def afterAll(): Unit = {
+    // Stops the testKit
+    TestKit.shutdownActorSystem(system)
   }
 
-  class FailingThenSucceedingTestDb(testProbe: ActorRef) extends Actor {
-
-    def counter(numberOfTrials: Int): Receive = {
-      case DbActor.WriteAndPropagate(channel, message) =>
-        if (numberOfTrials == 0) {
-          testProbe ! (channel, message)
-          sender() ! Status.Failure(DbActorNAckException(ErrorCodes.INVALID_ACTION.id, s"channel '$channel' does not exist in db"))
-          context.become(counter(numberOfTrials + 1))
-        } else {
-          testProbe ! (channel, message)
-          sender() ! DbActor.DbActorAck()
-          context.become(counter(0))
-        }
+  val input_file = "src/test/scala/util/examples/json/get_messages_by_id_answer.json"
+  val source: BufferedSource = scala.io.Source.fromFile(input_file)
+  val lines: String =
+    try source.mkString
+    finally source.close()
+  val getMessagesByIdResponse: JsonRpcResponse = JsonRpcResponse.buildFromJson(lines)
+
+  test("get_messages_by_id response should be processed without errors") {
+    val handler = GetMessagesByIdResponseHandler.graph(pubSubMediatorRef, MessageRegistry())
+    val output = Source.single(Right(getMessagesByIdResponse)).via(handler).runWith(Sink.head)
+
+    Await.ready(output, duration).value.get match {
+      case Success(Right(_)) => 0 should equal(0)
```

Comment:
You can use `Matchers.succeed` for unconditional success. On the other hand, you could check that the `ResultObject` has result 0.

Reply:
Unfortunately `Matchers.success` doesn't exist as far as I know